### PR TITLE
Add tests to travis CI and inject credentials in the test_credentials.json file

### PR DESCRIPTION
**Pull request by kchitalia**
@wmathurin @bhariharan Right now only the SalesforceSDKCommon and SalesforceSecurity tests pass. The other tests are failing. We can create bugs for the other tests and fix them. Since all the tests don't pass, travis CI will fail the build.

### CODE REVIEW

File: build/pre-build [hunk 1]
```diff
@@ -0,0 +1,12 @@
+#!/bin/bash
+
+echo "{
```
Comment (wmathurin) [inline build/pre-build]: :+1:

**Comment by wmathurin** [comment]
Looks good.

**Comment by kchitalia** [comment]
Please dont merge this pull request in, I am seeing unexpected behavior on the travis CI build. Its not picking up the environment variables I am using in the pre-build script. The build hangs/freezes on the RestAPI Tests. Maybe its a better idea to fix the tests first and then merge this piece in.

**Comment by kchitalia** [comment]
@wmathurin @bhariharan Removed the test schemes for which the tests are failing. Travis CI should pass the build now. Still need to figure out a way to inject test credentials.

**Comment by bhariharan** [comment]
:+1:

**Comment by kchitalia** [comment]
No need for this PR anymore

**Pull request closed without merging**